### ISSUE TITLE
As a user, I want to be able to edit in-progress amendments/revisions via the Forms Updated section, so that I can input changes

### ISSUE THREAD

**Issue by BCerki** [bcgov/cas-cif#1284]
#### Description:

NOTE: This ticket is outdated, check wireframes before refining. A tasklist-based editing workflow is here:https://github.com/bcgov/cas-cif/issues/1561

In this ticket, we change the workflow for editing amendments/general and minor revisions. This first step is to remove the "edit" button from forms. As a result, the tasklist can only be used to view forms. It can no longer be used to access or navigate editing functionality (with the exception of routinely updated fields, which always appear on the view page).

To edit amendments/general and minor revisions, a user must navigate to the desired project in the holistic change table and click the "View/Edit" button. After clicking the button, the user arrives at this page:
![image](https://user-images.githubusercontent.com/77306764/216691674-87027f27-0d76-40b8-8be8-7fdb2656ed73.png)
![image](https://user-images.githubusercontent.com/77306764/216691733-a257cf45-93ea-4b29-8c01-576ecf2557ab.png)

To access an in-progress form to make changes, users must click on the form's title in the "Forms Updated" section, which links to the form. The fields that don't apply to a particular revision type should be greyed out, but otherwise the form is editable.

If the user clicks on the tasklist at any point while working on an in-progress form, they're ejected from edit mode and sent to view the latest committed revision of that form. To get back to editing that form, they have to refind the project in the holistic change log and click "View/Edit" and the form's title again.

Outstanding questions:
- We talked about removing the tasklist from beside the holistic change table, decision?
- If a user wants to edit two of the amendment forms, they'll have to exit the project to change forms. Do we want that?
- We talked about removing the "Choose Forms to Be Updated" section. Does this mean at one point we're going to have no navigation?

Old context:

**After making a revision of any type, the "Amendments & Other Revisions" tab in the task list disappears. To continue with updates or check that the changes were captured, a user has to exit the project, then re-find it in the project list, which is inconvenient.

Keeping in mind the desired functionality in these issues:
-https://app.zenhub.com/workspaces/climate-action-secretariat-60ca4121764d710011481ca2/issues/gh/bcgov/cas-cif/1119
-https://app.zenhub.com/workspaces/climate-action-secretariat-60ca4121764d710011481ca2/issues/gh/bcgov/cas-cif/1122

[Envisioned workflow (prototype)](https://www.figma.com/proto/PzOY8RyQnVXnjZJdSv2AqQ/CIF---Internal-User-Portal?page-id=2289%3A2663&node-id=3169%3A2715&viewport=-104%2C637%2C0.11&scaling=scale-down&starting-point-node-id=3169%3A2715&show-proto-sidebar=1)**

#### Acceptance Criteria:

Given I am a user and I navigate to any of the forms using the tasklist
When I view the form
Then I do not see an edit button

Given I am a user and I navigate away from a revision and return to it (by clicking "View/Edit")
When I click on any of the form titles in the "Forms Updated" section
Then I am taken to the form
And the fields that cannot be updated via a revision of amendment type are greyed out 
And beside those fields is an explanation for why they are greyed out

Given I am a user and I look at the tasklist
When I click any of the links
I am ejected from edit and sent to view
And I see the appropriate routinely updated fields

#### Development Checklist:

- [ ] Checklist item
- [ ] Checklist item
- [ ] Checklist item
- [ ] Meets the DOD

**Definition of Ready** (Note: If any of these points are not applicable, mark N/A)

- [ ] User story is included
- [ ] User role and type are identified
- [ ] Acceptance criteria are included
- [ ] Wireframes are included (if required)
- [ ] Design / Solution is accepted by Product Owner
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

·**Definition of Done** (Note: If any of these points are not applicable, mark N/A)

- [ ] Acceptance criteria are tested by the CI pipeline
- [ ] UI meets accessibility requirements
- [ ] Configuration changes are documented, documentation and designs are updated
- [ ] Passes code peer-review
- [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
- [ ] Ticket is ready to be merged to main branch
- [ ] Can be demoed in Sprint Review
- [ ] Bugs or future work cards are identified and created
- [ ] Reviewed and approved by Product Owner

#### Notes:

-

**Comment by LindsayMacfarlane** [comment]
Hi @nanyangpro  @suhafa, Is there any existing design work that supports this ticket? Happy to walk through and discuss.

**Comment by nanyangpro** [comment]
Thanks for confirming, @LindsayMacfarlane! Yes, we **may** already have the user story covered in one of the existing prototypes. I just need a bit of clarification/context of the ticket itself, and would appreciate a quick chat before dropping the link. Would you recommend me reach out to @BCerki directly or call you instead?

**Comment by LindsayMacfarlane** [comment]
@nanyangpro - you can reach out to me.

**Comment by nanyangpro** [comment]
Wonderful, thanks Lindsay! If you are available, I'll call you after our Dev/Design Sync today (with a 5-min break for both of us :).

**Comment by nanyangpro** [comment]
Appreciate for the walkthrough, @LindsayMacfarlane! As discussed, the link to the [envisioned workflow (prototype)](https://www.figma.com/proto/PzOY8RyQnVXnjZJdSv2AqQ/CIF---Internal-User-Portal?page-id=2289%3A2663&node-id=3169%3A2715&viewport=-104%2C637%2C0.11&scaling=scale-down&starting-point-node-id=3169%3A2715&show-proto-sidebar=1) has been dropped in the ticket description. Thank you!

**Comment by BCerki** [comment]
@nanyangpro , where does the "New Revision" link go? And is the link text always "New Revision" (vs. Add, Edit, etc.)?

**Comment by nanyangpro** [comment]
Thanks for your question, @BCerki!

- It goes to the "New Revision" page for user to select "Revision Type", "Amendment Type" if "Revision Type" is Amendment, and forms to be updated. Please see the Envisioned workflow (prototype) in the ticket description.
- Yes.

**Comment by BCerki** [comment]
Thanks, @nanyangpro ! A couple follow-up questions:

1. do we send the user to the "New Revision" page even if they're returning to a revision they've already created? (Ie, the already chose the forms to be updated and already clicked the "new revision" button).
2. how does the user get to the page with pending actions from and the revision record history?

**Comment by BCerki** [comment]
To clarify the questions, when the user clicks the "New Revision" link in the tasklist:
![image](https://user-images.githubusercontent.com/77306764/213001806-a2b10760-d284-4c41-a122-70cee0109da3.png)
They should be redirected where? I'm guessing it's one of  these two pages, or possibly conditional depending on what stage the revision is at:

1. ![image](https://user-images.githubusercontent.com/77306764/213001371-c39c01b5-d124-4205-aa6e-35cf15221f0a.png) (should a user ever see this page except for when they're creating the revision for the first time?)

1. ![image](https://user-images.githubusercontent.com/77306764/213001624-c10cb787-306d-4030-a248-744fa714624d.png) (currently the only way to get to this page is to exit the revision and then come back into it through the  "View/Edit" button in the table, see [Loom](https://www.loom.com/share/6ba35ea2b0f84a81a6d0c7d20961e43a) with audio)

Sidenote, this is what currently happens if you click the "New Revision" button if there's already a revision: https://www.loom.com/share/532da8abe2e24ce38c170db6c658d05e

@Sepehr-Sobhani , @gurjmatharu , we've all had this workflow come up in PRs lately; feel free to add to or edit this comment.

**Comment by Sepehr-Sobhani** [comment]
Just some context and sharing my thoughts :)
We don't create a new revision/amendment before hitting `New Revision` button, and after hitting the button we are in the edit mode and there's an ongoing revision so maybe the most straightforward way of having access to the revisions/amendments table is to bringing it back in the tasklist or maybe having a button that takes the user directly to the revision/amendment edit/view page.

**Comment by gurjmatharu** [comment]
> Just some context and sharing my thoughts :) We don't create a new revision/amendment before hitting `New Revision` button, and after hitting the button we are in the edit mode and there's an ongoing revision so maybe the most straightforward way of having access to the revisions/amendments table is to bringing it back in the tasklist or maybe having a button that takes the user directly to the revision/amendment edit/view page.

Also when you're editing a revision you can't edit another revision so even if you have access to the list you won't be able to switch editing revisions.

**Comment by BCerki** [comment]
> Just some context and sharing my thoughts :) We don't create a new revision/amendment before hitting `New Revision` button, and after hitting the button we are in the edit mode and there's an ongoing revision so maybe the most straightforward way of having access to the revisions/amendments table is to bringing it back in the tasklist or maybe having a button that takes the user directly to the revision/amendment edit/view page.

To make sure I've got this straight since there are multiple `New Revision` buttons:
-the button on the Amendments and Other Revisions page above the table is just a redirect; it doesn't actually create the revision
- the button on the page with "choose forms to be updated" (labelled page 1 above) is what creates the revision

So I think that means the user can never get back to page 1 once they've created a revision. (Similar to how when you first create a project you get to the select rfp form, and once you've clicked create project, you can never get back to the select rfp form again.) So that means that after clicking `New Revision` on page 1, the user can never change amendment type general/minor/amendment or scope/cost/schedule?

**Comment by Sepehr-Sobhani** [comment]
- Regarding the button, yes that's true, the button on page that creates the new revision.
- With the current approach, yes this is the flow. when we hit that button the revision is created and after that, we can't change the amendment type.

**Comment by BCerki** [comment]
This could be split into separate tickets for each form. A few more questions:
- how does the tasklist behave when you're editing a form with greyed-out fields

**Comment by Dianadec** [comment]
Next steps from Feb. 7 backlog refinement:
- Setup dev design sync (maybe 30 mins?)
- Setup meeting with CIF Ops team.
Goal is by end of this sprint (by Feb 17).

**Comment by LindsayMacfarlane** [comment]
Backlog refinement (Feb 22, 2022): There are still a lot of outstanding questions to support moving forward with amendments and revisions. Taking a pause to do a more in-depth review/whiteboarding session before continuing with the implementation of this EPIC.

**Comment by pbastia** [comment]
this might still be relevant after #1523 is completed. unless @nanyangpro do you have an answer for that?

**Comment by pbastia** [comment]
Not needed at this point, we'll create a new card with updated prototypes when needed.
